lib/compilers/nim.ts

```diff
@@ -106,7 +106,21 @@ export class NimCompiler extends BaseCompiler {
                 filters.binary = true;
                 const objFile = unwrap(this.getCacheFile(options!, result.inputFilename!, cacheDir));
                 if (await utils.fileExists(objFile)) {
-                    await fs.rename(objFile, outputFilename);
+                    try {
+                        // First try rename (fastest), fall back to copy+delete if it fails
+                        await fs.rename(objFile, outputFilename);
+                    } catch (renameError) {
+                        // Rename can fail across filesystems, so copy and delete as fallback
```

Comment:
Can I ask why this complex behaviour has been added? Is there any evidence this has happened ever? We compute the filenames from each other, they usually differ in extension only. So they are in the same directory.

Please don't add extra complexity where no such issue exists.

Conversely _if_ this was a real issue we would make a utility function to do this and call it consistently across all the codebase (this is a very common idiom).